### PR TITLE
media-sound/musepack-tools: fix build on musl

### PR DESCRIPTION
**Pull request by cfuga**
Signed-off-by: Cristian Othón Martínez Vera <cfuga@cfuga.mx>
Bug: https://bugs.gentoo.org/712978

This patch makes the following changes.
* __linux__ isn't the same as __GLIBC__

Thus, it avoids calling <fpu_control.h> macros in musl, doesn't require
conditional patching and still works nicely with glibc-based systems.

* Includes <sys/select.h> for select(3). That header includes <sys/time.h>
and also fixes the struct timeval error.

### CODE REVIEW

File: media-sound/musepack-tools/files/musepack-tools-465-musl.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+diff -uNr a/mpcenc/keyboard.c b/mpcenc/keyboard.c
```
Comment (thesamesam) [inline media-sound/musepack-tools/files/musepack-tools-465-musl.patch:1]: - Please run scrub-patch from iwdevtools
- Please send upstream if possible and include the link at the top of the patch
- Please include links to any relevant Gentoo bugs at the top of the patch

**Comment by thesamesam** [comment]
Thank you!

**Comment by cfuga** [comment]
Finally, I could access the upstream repo, and I found the patches for musl were applied in 2016.
Should I replace the current musl patch with the original patches from the SVN repo?

**Comment by thesamesam** [comment]
Yes please, but in a new PR. Merged in c0c3f271f5d35a31cfb5225885aaa5258d471590 (squashed so I lost the Closes tag). Thank you!

**Pull request closed without merging**